bibliofil_lists.module

```diff
@@ -0,0 +1,293 @@
+
+  if ($result->code != 200) {
+    watchdog('bibliofil_lists', 'There was problem on requesting API.', NULL, WATCHDOG_ERROR);
+  }
+
+  $data = drupal_json_decode($result->data);
+
+  $tabs = [];
+  if (!empty($conf_list_type)) {
+    if (!empty($data[$conf_list_type])) {
+      foreach ($data[$conf_list_type] as $key => $item) {
+        $tabs[$key]['#title'] = $item['name'];
+        $tabs[$key]['#theme'] = 'ding_carousel';
+        $tabs[$key]['#placeholders'] = '';
+        $tabs[$key]['#offset'] = 0;
+        $tabs[$key]['#path'] = '#';
+
+        foreach ($item['titler'] as $k => $i) {
+          $tabs[$key]['#items'][] = [
+            '#markup' => theme('bibliofil_lists_item', [
+              'type' => 'suggestions',
+              'opac_url' => $opac_url,
+              'data' => $i,
+            ]),
+          ];
+        }
+      }
+    }
+  }
+  else {
+    $nonempty = array_filter($data, function ($category) {
+      if (is_array($category)) {
+        return $category;
+      }
+    });
+
+    $inlined = [];
+    foreach ($nonempty as $type) {
+      foreach ($type as $t) {
+        $inlined[] = $t;
+      }
+    }
+
+    foreach ($inlined as $key => $tabgen) {
+      $tabs[$key]['#title'] = $tabgen['name'];
+      $tabs[$key]['#theme'] = 'ding_carousel';
+      $tabs[$key]['#placeholders'] = '';
+      $tabs[$key]['#offset'] = 0;
+      $tabs[$key]['#path'] = '#';
+
+      foreach ($tabgen['titler'] as $k => $i) {
+        $tabs[$key]['#items'][] = [
+          '#markup' => theme('bibliofil_lists_item', [
+            'type' => 'suggestions',
+            'opac_url' => $opac_url,
+            'data' => $i,
+          ]),
+        ];
+      }
+    }
+  }
+
+  if (!empty($tabs)) {
+    return $element = [
+      '#theme' => 'ding_tabbed_carousel',
+      '#transition' => 'fade',
+      '#tabs' => $tabs,
+      '#attached' => [
+        'js' => [drupal_get_path('module', 'ding_carousel') . '/js/ding_carousel.js'],
+      ],
+    ];
+  }
+}
+
+/**
+ * Request and build library lists carousel.
+ *
+ * @param string $hash
+ *   Request hash.
+ *
+ * @return array|null
+ *   Renderable array if data and empty array if no data.
+ *
+ * @throws \Exception
+ */
+function bibliofil_lists_build_librarylists($hash) {
+  $element = [];
+  $opac_url = variable_get('bibliofil_lists_opac_url', '');
+
+  $decoded = base64_decode($hash);
+  $explode = explode(':', $decoded);
+  $conf_random = $explode[0];
+  $conf_instans = $explode[1];
+
+  $query_params = [
+    'mode' => 'show',
+    'json' => 1,
+    'showlibrarylist' => 1,
+    'showpatronlist' => 1,
+    'random' => $conf_random,
+    'instans' => $conf_instans,
+  ];
+
+  $url = url($opac_url . WEBLISTERPATH, ['query' => $query_params]);
+
+  $result = drupal_http_request($url);
+
+  if ($result->code != 200) {
+    watchdog('bibliofil_lists', 'There was problem on requesting API.', NULL, WATCHDOG_ERROR);
+  }
+
+  $data = drupal_json_decode($result->data);
```

Comment:
I guess that if the result code was not 200 there's no point to decode anything. Therefore I think the function should exit early.

Reply:
\+